kernel/setup/steps/ezstep_installer.php

```diff
@@ -489,12 +489,11 @@ function checkDatabaseRequirements( $dbCharset = false, $overrideDBParameters =
             $checkedCharset = $db->checkCharset( $charsetsList, $currentCharset );
             if ( $checkedCharset === false )
             {
-                // If the current charset is utf-8 we use that instead
-                // since it can represent any character possible in the chosen languages
-                if ( $currentCharset == 'utf-8' )
+                // If the current charset is utf-8 or utf8mb4 we use that instead
+                // since they can represent any character possible in the chosen languages
+                if ( in_array( $currentCharset, array( 'utf-8', 'utf8mb4' ) ) )
                 {
-                    $charset = 'utf-8';
-                    $result['site_charset'] = $charset;
+                    $result['site_charset'] = $currentCharset;
```

Comment:
```suggestion
                    $result['site_charset'] = 'utf-8';
```

As this option is in the end passed on to `i18n.ini` it seems like it might need to be set to utf-8 even if we got utf8mb4 as input. Or do we need to adapt the whole setup code base to support this value?

Reply:
@andrerom Right, we only need `utf8mb4` when talking to mysql. We can test it.